### PR TITLE
Severity limit for rules

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Add "--severity" key to skip rules with severity under the limit
- Update test with help according python3.10 argparse

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest

### CODE REVIEW

File: credsweeper/app.py [hunk 1]
```diff
@@ -72,14 +73,23 @@ def __init__(self,
             find_by_ext: boolean - files will be reported by extension
             depth: int - how deep container files will be scanned
             doc: boolean - document-specific scanning
+            severity: Severity - minimal severity level of rule
```
Comment (csh519) [inline credsweeper/app.py:76]:
```suggestion
            severity: Severity - minimum severity level of rule
```

File: credsweeper/app.py [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
 import sys
 from pathlib import Path
-from typing import Any, List, Optional, Union
+from typing import Any, List, Optional, Union, Dict
 
 import pandas as pd
 
 from credsweeper.app_path import APP_PATH
-from credsweeper.common.constants import KeyValidationOption, ThresholdPreset
+from credsweeper.common.constants import KeyValidationOption, ThresholdPreset, Severity
```
Comment (csh519) [inline credsweeper/app.py:12]:
```suggestion
from credsweeper.common.constants import KeyValidationOption, Severity, ThresholdPreset
```

File: credsweeper/app.py [hunk 3]
```diff
@@ -95,40 +105,43 @@ def __init__(self,
...
+        if severity:
+            config_dict["severity"] = severity.value
 
         if exclude_lines is not None:
             config_dict["exclude"]["lines"] = config_dict["exclude"].get("lines", []) + exclude_lines
         if exclude_values is not None:
             config_dict["exclude"]["values"] = config_dict["exclude"].get("values", []) + exclude_values
 
-        return config_dict
+        return config_dict  # type: ignore
```
Comment (csh519) [inline credsweeper/app.py:144]: `type: ignore`? What does it mean?

File: credsweeper/app.py (hunk 2 above)
Comment (babenek) [reply credsweeper/app.py:12]: What reason of the swap? Alphabet order only?

File: credsweeper/app.py (hunk 3 above)
Comment (babenek) [reply credsweeper/app.py:144]: MYPY warns about returning Any. I've suggested this suppression in line will be better that full suppresion in config.

**Review by csh519 — DISMISSED** [review]
Thank you for adding `Severity` related features.

LGTM :+1:

**Review by xDizzix — APPROVED** [review]
Severity filtering is a good idea

File: credsweeper/__main__.py [hunk 4]
```diff
@@ -68,6 +68,21 @@ def logger_levels(log_level: str) -> str:
     raise ArgumentTypeError(f"log level given: {log_level} -- must be one of: {' | '.join(Logger.LEVELS.keys())}")
 
 
+def severity_levels(severity_level: str) -> Severity:
+    """Severity level correctness verification and transformation
+
+    Args:
+        severity_level: string with level
+
+    Returns True if log_level UPPERCASE is one of keys
```
Comment (kmnls) [inline credsweeper/__main__.py:77]: Wrong description. This is a copy-paste error

File: credsweeper/__main__.py [hunk 5]
```diff
@@ -68,6 +68,21 @@ def logger_levels(log_level: str) -> str:
...
+    """
+
+    if severity := Severity.get(severity_level):
+        return severity
+    raise ArgumentTypeError(
+        f"Severity level given: {severity_level} -- must be one of: {' | '.join([i.value for i in Severity])}")
```
Comment (kmnls) [inline credsweeper/__main__.py:83]:
```suggestion
        f"Severity level provided: {severity_level} -- must be one of: {' | '.join([i.value for i in Severity])}")
```

File: credsweeper/__main__.py [hunk 6]
```diff
@@ -109,6 +124,12 @@ def get_arguments() -> Namespace:
                         default=None,
                         dest="rule_path",
                         metavar="PATH")
+    parser.add_argument("--severity",
+                        help=f"set minimum level for rules {[i.value for i in Severity]}"
```
Comment (kmnls) [inline credsweeper/__main__.py:128]: "set minimum level for rules to apply" ?

File: docs/source/guide.rst [hunk 7]
```diff
@@ -28,6 +32,7 @@ Get all argument list:
     --export_log_config [PATH]
                             exporting default logger config to file (default: log.yaml)
     --rules [PATH]        path of rule config file (default: credsweeper/rules/config.yaml). severity:['critical', 'high', 'medium', 'low', 'info'] type:['keyword', 'pattern', 'pem_key']
+    --severity SEVERITY set minimum level for rules ['critical', 'high', 'medium', 'low', 'info'](default: 'Severity.INFO', case insensitive)
```
Comment (kmnls) [inline docs/source/guide.rst:35]:
```suggestion
    --severity SEVERITY    set minimum level for rules ['critical', 'high', 'medium', 'low', 'info'](default: 'Severity.INFO', case insensitive)
```

**Review by kmnls — APPROVED** [review]
Agree with the feature implementation but some comments need improvement

File: credsweeper/__main__.py (hunk 6 above)
Comment (babenek) [inline credsweeper/__main__.py:128]:
```suggestion
                        help=f"set minimum level for rules to apply {[i.value for i in Severity]}"
```

**Review by iuriimet — APPROVED** [review]

**Review by xDizzix — APPROVED** [review]
+

**Review by kmnls — APPROVED** [review]
Agree

**Pull request merged**